src/caffe/test/test_inner_product_layer.cpp

```diff
@@ -148,4 +265,127 @@ TYPED_TEST(InnerProductLayerTest, TestGradient) {
   }
 }
 
+TYPED_TEST(InnerProductLayerTest, TestGradientTransposeFalse) {
```

Comment:
Shouldn't this test be `TestGradientTransposeTrue` (with the corresponding change from `set_transpose(false)` to `set_transpose(true)`)?  This test is effectively a duplicate of the existing test above (TestGradient), I'd think.

And if this test is changed to be done with `transpose` on, is there still anything additionally tested by `TestBackwardTranspose`?  I would think the combination of `TestForward` with the gradient check would cover all functionality.